proposals/3/README.md

```diff
@@ -0,0 +1,201 @@
+
+
+# 2. Design
+
+
+## 2.1 UX
+
+For the user, the proposed change means that he is responsible for creating all required cloud resources up front (prior to installing Jenkins X).
+We provide documentation on what’s required as well as Terraform scripts to create and manage the required resources. 
+
+When using Terraform, `terraform apply` is effectively replacing current `jx create`.
+
+If a Jenkins X feature requires new Cloud resources, upgrading the cluster will fail until the user has created the new required resource, either manually or via the Terraform scripts. 
+
+The Jenkins X install process will always only verify whether a needed resource exists, never create it
+When verification fails, the install process halts and needs to provide enough context to the user to identify which (cloud) resource is missing.
+
+
+## 2.2. Technical Design
+
+
+### 2.2.1 Docs
+
+Ar part of this work we want to clean up and extend the documentation.
+The updated documentation should include sections for the following:
+
+
+
+*   Generic description of the cloud resources needed and their purpose
+*   Documentation on where to find the Terraform script and how to use them
+
+
+### 2.2.2 Terraform
+
+As part of the work, we are creating a GitHub repository with two Terraform setups.
+One for GKE and one for EKS.
+In each case, it is the responsibility of the Terraform script to create the following (cloud) resources:
+
+
+
+*   Required service accounts
+    *   _externaldns-sa_
+        *   Optional, only enabled if user wants to use external DNS
+        *   IAM roles:
+            *   GKE - roles/iam.workloadIdentityUser, roles/dns.admin
+            *   EKS - IAM/role and IAM Policy with Route53 permissions., Kubernetes Service Account.
+        *   
+    *   _kaniko-sa_
+        *   required
+        *   IAM roles:	
+            *   GKE - roles/iam.workloadIdentityUser, roles/storage.admin, roles/storage.objectAdmin, roles/storage.objectCreator
+            *   EKS - Covered by the tekton-bot Service Account.
+    *   _storage-sa_
+        *   required
+        *   IAM roles:	
+            *   GKE - roles/iam.workloadIdentityUser, storage.objects.[create|get|delete|update]
+            *   EKS - S3Access Policy attached to an IAM/role and a Kubernetes Service Account.
+    *   _tekton-sa_
+        *   required
+        *   IAM roles:	
+            *   GKE - roles/iam.workloadIdentityUser, roles/viewer 
+            *   EKS - IAM/role and IAM Policy with general permissions, Kubernetes Service Account.
+    *   _velero-sa_
+        *   Optional, only created if the user wants to use Velero
+        *   IAM roles:	
+            *   GKE - roles/iam.workloadIdentityUser, roles/storage.admin, roles/storage.objectAdmin, roles/storage.objectCreator
+            *   EKS - S3Access Policy attached to an IAM/role and a Kubernetes Service Account.
+*   Storage buckets 
```

Comment:
Could we just use one bucket? Then we would get around the need to create new ones as the user wants to try new features (eg Velero, bucketrepo, etc). At least the Terraform setup could use a single bucket.